__tests__/posts.ts

```diff
@@ -2138,3 +2141,207 @@ describe('mutation checkLinkPreview', () => {
+  const MUTATION = `
+    mutation EditPost($id: ID!, $title: String, $content: String, $image: Upload) {
+      editPost(id: $id, title: $title, content: $content, image: $image) {
+        _
+      }
+    }
+  `;
+
+  const params = {
+    id: 'p1',
+    title: 'This is a welcome post',
+  };
+
+  beforeEach(async () => {
+    await con
+      .getRepository(Source)
+      .update({ id: 'a' }, { type: SourceType.Squad });
+    await con
+      .getRepository(Post)
+      .update(
+        { id: 'p1' },
+        { type: PostType.Welcome, title: 'Welcome post', authorId: '1' },
+      );
+    await con.getRepository(SourceMember).save([
+      {
+        userId: '1',
+        sourceId: 'a',
+        role: SourceMemberRoles.Member,
+        referralToken: randomUUID(),
+      },
+      {
+        userId: '2',
+        sourceId: 'a',
+        role: SourceMemberRoles.Member,
+        referralToken: randomUUID(),
+      },
+    ]);
+  });
+
+  it('should not authorize when not logged in', () =>
+    testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: params },
+      'UNAUTHENTICATED',
+    ));
+
+  it('should return an error if post type is not allowed to be editable', async () => {
+    loggedUser = '1';
+    await con
+      .getRepository(Post)
+      .update({ id: 'p1' }, { type: PostType.Share });
+
+    await testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: params },
+      'FORBIDDEN',
+    );
+
+    await con
+      .getRepository(Post)
+      .update({ id: 'p1' }, { type: PostType.Article });
+
+    return testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: params },
+      'FORBIDDEN',
+    );
+  });
+
+  it('should return an error if title exceeds 80 characters', async () => {
+    loggedUser = '1';
+
+    return testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: {
+          ...params,
+          title:
+            'Hello World! Start your squad journey here - Hello World! Start your squad journey here',
+        },
+      },
+      'GRAPHQL_VALIDATION_FAILED',
+    );
+  });
+
+  it('should return an error if content exceeds 4000 characters', async () => {
+    loggedUser = '1';
+
+    const content = 'Hello World! Start your squad journey here';
+    const sample = new Array(100).fill(content);
+
+    return testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: {
+          ...params,
+          content: sample.join(' - '),
+        },
+      },
+      'GRAPHQL_VALIDATION_FAILED',
+    );
+  });
+
+  it('should restrict member when user is not the author of the post', async () => {
+    loggedUser = '2';
+
+    return testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: { id: 'p1', title: 'Test' } },
+      'FORBIDDEN',
+    );
+  });
+
+  it('should not update title if the post type is "welcome"', async () => {
```

Comment:
Are you sure welcome post title cannot be edited?

Reply:
For the v5 iteration, it should not be the case yet.